src/pages/programs/index.vue

```diff
@@ -2,71 +2,89 @@
   <div v-if="loading">
     <Loading />
   </div>
-  <div class="flex flex-col w-full h-full">
-    <div class="flex w-full p-6 mb-4 text-lg bg-white border-b border-solid shadow-sm">
-      <heroicons-outline-search class="mr-4 text-2xl text-gray-500" />
-      <input
-        v-model="searchQuery"
-        class="w-full text-lg outline-none "
-        placeholder="Rechercher"
-      >
-    </div>
 
-    <div class="flex flex-col flex-grow p-6 space-y-3">
-      <div class="flex items-center justify-between border-b">
-        <div class="flex items-center">
-          <h1 class="mb-2 text-3xl">
-            Programmes
-          </h1>
-          <Badge class="mb-1 ml-4 text-sm bg-gray-500">
-            {{ filteredPrograms.length }}
-          </Badge>
-        </div>
-        <div class="p-2 mb-2 text-white bg-green-500 rounded-full shadow cursor-pointer hover:bg-green-400">
-          <heroicons-outline-plus class="text-xl" />
-        </div>
-      </div>
+  <AppLayout>
+    <template #title>
+      Programs
+    </template>
+
+    <template #actions>
+      <AppButton :color="allProgramsAreStopped() ? 'green' : ''" @click="startPrograms()">
+        <heroicons-outline-arrow-circle-up class="mr-1" />
+        {{ allProgramsAreStopped() ? "Start" : "Restart" }} all
```

Comment:
To benefit from Vue's reactivity, you may want to use a computed property instead of a method.
The method will not be automatically rerun when its reactive dependencies change, with a computed property it would.